### PR TITLE
Tps command

### PR DESCRIPTION
**Pull request by Trompettesib**

### CODE REVIEW

File: cubic-server/Dimension.cpp [hunk 1]
```diff
@@ -9,22 +9,38 @@
...
     _isInitialized(false),
     _isRunning(false),
-    _dimensionType(dimensionType)
+    _level(),
+    _loadingChunks({}),
+    _processingThread(),
+    _dimensionType(dimensionType),
+    _circularBufferTps(18000),
+    _previousTickTime(std::chrono::high_resolution_clock::now()),
+    _circularBufferMSPT(18000)
```
Comment (huntears) [inline cubic-server/Dimension.cpp:37]: Please don't use magic numbers here

File: cubic-server/Dimension.cpp [hunk 2]
```diff
@@ -330,3 +354,45 @@ void Dimension::sendChunkToPlayers(int x, int z)
     }
     this->_loadingChunks.erase({x, z});
 }
+
+Tps Dimension::getTps() const
+{
+    const auto &buffer_size = _circularBufferTps.size();
+    const auto &buffer_end = _circularBufferTps.end();
```
Comment (huntears) [inline cubic-server/Dimension.cpp:361]: No need to get references here as it is a size and an iterator.

File: cubic-server/Dimension.cpp [hunk 3]
```diff
@@ -330,3 +354,45 @@ void Dimension::sendChunkToPlayers(int x, int z)
...
+        return tps;
+    }
+    tps.fifteenMinTps = tpsCalculation(TICKS_FOR_FIFTEEN_MINUTES);
+    return tps;
+}
+
+MSPTInfos Dimension::getMSPTInfos() const
+{
+    const auto &buffer_begin = _circularBufferMSPT.begin();
+    const auto &buffer_end = _circularBufferMSPT.end();
```
Comment (huntears) [inline cubic-server/Dimension.cpp:390]: No need to get references here as it is a size and an iterator.

File: cubic-server/command_parser/commands/MSPT.cpp [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+#include "logging/logging.hpp"
+#include <vector>
+
+using namespace command_parser;
+
+void MSPT::autocomplete(UNUSED std::vector<std::string> &args, UNUSED Player *invoker) const { }
+
+void MSPT::execute(std::vector<std::string> &args, Player *invoker) const
+{
+    switch (args.size()) {
```
Comment (huntears) [inline cubic-server/command_parser/commands/MSPT.cpp:17]: Please don't use a switch case for just 3 branching.
Please also put some of that code in other functions as the execute method is getting quite big.

File: cubic-server/command_parser/commands/Tps.cpp [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+{
+    if (invoker)
+        return;
+    else
+        LINFO("autocomplete time");
+}
+
+void Tps::execute(std::vector<std::string> &args, Player *invoker) const
+{
+    switch (args.size()) {
```
Comment (huntears) [inline cubic-server/command_parser/commands/Tps.cpp:25]: Please don't use a switch case for just 3 branching.
Please also put some of that code in other functions as the execute method is getting quite big.

File: cubic-server/types.cpp [hunk 6]
```diff
@@ -1,6 +1,5 @@
 #include "types.hpp"
 #include "utility/PseudoRandomGenerator.hpp"
-#include <cstdint>
```
Comment (huntears) [inline cubic-server/types.cpp:3]: Why? We use some of cstdint's types in that file...

File: cubic-server/types.hpp [hunk 7]
```diff
@@ -212,4 +212,18 @@ struct std::hash<Position2D> {
...
+struct Tps {
+    float oneMinTps;
+    float fiveMinTps;
+    float fifteenMinTps;
+    std::string toString() const;
+};
+
+struct MSPTInfos {
+    float mean;
+    float min;
```
Comment (huntears) [inline cubic-server/types.hpp:224]: You need to put the min first as in-game the min is higher than the average :)

Comment (huntears) [reply cubic-server/types.hpp:224]: See the screenshot:
image

**Review by huntears — CHANGES_REQUESTED** [review]

File: cubic-server/types.cpp (hunk 6 above)
Comment (Trompettesib) [reply cubic-server/types.cpp:3]: It's already included in types.hpp

File: cubic-server/Dimension.hpp [hunk 8]
```diff
@@ -15,6 +16,11 @@
 
 // TODO(huntears): Fix whatever this is
 constexpr int SEMAPHORE_MAX = 1000;
+constexpr int TICK_PER_MINUTE = 20 * 60;
+constexpr int TICKS_FOR_FIVE_MINUTES = TICK_PER_MINUTE * 5;
+constexpr int TICKS_FOR_FIFTEEN_MINUTES = TICK_PER_MINUTE * 15;
```
Comment (STMiki) [inline cubic-server/Dimension.hpp:21]: A simple calculation like this don't need to be constexpr, just use `TICK_PER_MINUTES`

**Review by STMiki — DISMISSED** [review]

File: cubic-server/types.cpp (hunk 6 above)
Comment (huntears) [reply cubic-server/types.cpp:3]: It doesn't matter, as we have include guards, but aight

**Review by huntears — APPROVED** [review]

**Review by STMiki — DISMISSED** [review]

**Review by STMiki — DISMISSED** [review]

**Review by PierrickMarguin — APPROVED** [review]

**Review by huntears — APPROVED** [review]

**Review by STMiki — APPROVED** [review]

**Pull request merged**